modules/dqt/jsx/react.app.js

```diff
@@ -1151,7 +1154,7 @@ class DataQueryApp extends Component {
    */
   updateFilter(filter) {
     if (filter.children.length === 0) {
-      filter.session = this.state.filter.session;
+      filter.session = this.state.allSessions;
```

Comment:
@ridz1208 this change is needed for "Deleting all filters causes query page to fail" and this.state.allSessions is a persistent collection of all the sessions that will never be modified. It's needed here in this case when all rules are deleted and the sessions need to be reset back to all sessions.